gammapy/modeling/serialize.py

```diff
@@ -106,13 +106,27 @@ def _link_shared_parameters(models):
                     shared_register[name] = param
 
 
-def datasets_to_dict(datasets, path, overwrite):
+def datasets_to_dict(datasets, path, prefix, overwrite):
+    """Convert datasets to dicts for serialization.
+
+    Parameters
+    ----------
+    datasets : `~gammapy.modeling.Datasets`
+        Datasets
+    path: `pathlib.Path`
+        path to write files
+    prefix : str
+        common prefix of file names
+    overwrite : bool
+        overwrite datasets FITS files
+    """
+
     unique_models = []
     unique_backgrounds = []
     datasets_dictlist = []
 
     for dataset in datasets:
-        filename = path + "data_" + dataset.name + ".fits"
+        filename = str(path / f"{prefix}_data_{dataset.name}.fits")
```

Comment:
Usually, we use the coding pattern to use `Path` objects as much as possible, and then to only put `str(path)` where needed, i.e. when you put it in a dict for serialisation, or when you call an external API that doesn't support Path yet.
So suggest to remove the `str` here.

https://dev.azure.com/gammapy/gammapy/_build/results?buildId=2112&view=logs&jobId=d7d66435-5ae4-561a-5876-2cd4e19b3ce2&taskId=922dfe32-5b90-53bc-18db-0fa0ecbbd97b&lineStart=199&lineEnd=200&colStart=1&colEnd=1